evm/src/generation/prover_input.rs

```diff
@@ -251,38 +251,39 @@ impl<F: Field> GenerationState<F> {
 
     /// Returns the next used jump address.
     fn run_next_jumpdest_table_address(&mut self) -> Result<U256, ProgramError> {
-        let context = self.registers.context;
-        let code_len = u256_to_usize(self.get_code_len()?.into());
+        let context = u256_to_usize(stack_peek(self, 0)? >> CONTEXT_SCALING_FACTOR)?;
```

Comment:
The test `test_jumpdest_analysis` also calls `jumpdest_analysis` with context 3 in the stack. Without this change it fails because the current context is 0 instead.

Reply:
Ah ok it's because the test has a special scenario, but you cannot reach `jumpdest_analysis` (outside of this test) without calling `enter_new_ctx`. But yeah, harmless change anyway